crates/bevy_reflect/src/impls/glam.rs

```diff
@@ -1,44 +1,195 @@
 use crate as bevy_reflect;
+use crate::reflect::Reflect;
 use crate::ReflectDeserialize;
-use bevy_reflect_derive::{impl_from_reflect_value, impl_reflect_value};
+use bevy_reflect_derive::{
+    impl_from_reflect_value, impl_reflect_struct_and_from_reflect_struct, impl_reflect_value,
+};
 use glam::*;
 
-impl_reflect_value!(IVec2(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(IVec3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(IVec4(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(UVec2(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(UVec3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(UVec4(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Vec2(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Vec3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Vec3A(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Vec4(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(DVec2(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(DVec3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(DVec4(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Mat3(PartialEq, Serialize, Deserialize));
-impl_reflect_value!(Mat4(PartialEq, Serialize, Deserialize));
+impl_reflect_struct_and_from_reflect_struct!(
+    Constructor(Default::default())
+    BevyReflectPath(self::bevy_reflect)
```

Comment:
I'm not sure I understand what this is for. Wouldn't this be the same as using `BevyManifest`? Or is it doing something different?

Reply:
I ran into a strange issue where specifically `use bevy_reflect::Struct` is ambiguous when called from a context where `use crate as bevy_reflect` is defined. This was not a problem before since we're only defining a Value reflect, but for a Struct reflect it needs to be either `self::bevy_reflect` or `crate::` to work, so I needed some way to override the path. I felt like this was a less intrusive option that messing with BevyManifest or infecting more macros than I already had to touch. If you have a better suggestion I would love to hear it! Not a big fan of either the Constructor() or BevyReflectPath() parts, but wasn't sure how best to get around it.

Reply:
Hm, that's strange. I wonder why that's the case. So does it only affect the test modules (which is where I'm assuming `use crate as bevy_reflect` is used)?

And yeah, I think `#[reflect(path = "self::bevy_reflect", ctor = "Default::default()"]` is definitely an improvement. It's more consistent with the attributes we have now.

Reply:
yep! I'm working on implementing those both once I get to them. I do think ctor could get a little weird if it gets complicated but that's probably still better than the weird tag thing I made.
And no, this affects the glam module where the glam types are impl'd for reflect. I feel like I may simply be not getting something but hopefully someone else figures that out before this is merged because I could not. Would be nice to just drop the path attribute entirely